src/sqlfluff/rules/L005.py

```diff
@@ -38,19 +39,15 @@ class Rule_L005(BaseRule):
     """
 
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
-        """Commas should not have whitespace directly before them.
-
-        We need at least one segment behind us for this to work.
-
-        """
-        if len(context.raw_stack) >= 1:
-            cm1 = context.raw_stack[-1]
-            if (
-                context.segment.is_type("comma")
-                and cm1.is_type("whitespace")
-                and cm1.pos_marker.line_pos > 1
-            ):
-                anchor = cm1
-                return LintResult(anchor=anchor, fixes=[LintFix.delete(cm1)])
-        # Otherwise fine
+        """Commas should not have whitespace directly before them."""
```

Comment:
The primary change to this file is switching from `raw_stack` to `raw_segment_pre`, which is faster. Also did some tidying, but no other meaningful changes. @OTooleMichael